google-oauth-client/src/main/java/com/google/api/client/auth/openidconnect/IdTokenVerifier.java

```diff
@@ -142,9 +204,87 @@ public final String getIssuer() {
    * @return {@code true} if verified successfully or {@code false} if failed
    */
   public boolean verify(IdToken idToken) {
-    return (issuers == null || idToken.verifyIssuer(issuers))
-        && (audience == null || idToken.verifyAudience(audience))
-        && idToken.verifyTime(clock.currentTimeMillis(), acceptableTimeSkewSeconds);
+    boolean simpleChecks =
+        (issuers == null || idToken.verifyIssuer(issuers))
+            && (audience == null || idToken.verifyAudience(audience))
+            && idToken.verifyTime(clock.currentTimeMillis(), acceptableTimeSkewSeconds);
+
+    if (!simpleChecks) {
+      return false;
+    }
+
+    // This method validates token signature per current OpenID Connect Spec:
+    // https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation
+    // By default, method gets a certificate from well-known location
+    // A request to certificate location is performed using
+    // {@link com.google.api.client.http.javanet.NetHttpTransport}
+    // Both certificate location and transport implementation can be overridden via {@link Builder}
+    // not recommended: this check can be disabled with OAUTH_CLIENT_SKIP_SIGNATURE
+    // environment variable set to true.
+    try {
+      return verifySignature(idToken);
+    } catch (VerificationException ex) {
```

Comment:
Logging? All these good messages in the exception are ignored here.

Reply:
yes, probably makes sense